File: Examples/CSharp/AsposePdfFacades/Attachments/ExtractAllAttachments.cs
//////////////////////////////////////////////////////////////////////////
// Copyright 2001-2014 Aspose Pty Ltd. All Rights Reserved.
//
// This file is part of Aspose.Pdf. The source code in this file
// is only intended as a supplement to the documentation, and is provided
// "as is", without warranty of any kind, either expressed or implied.
//////////////////////////////////////////////////////////////////////////
using System.IO;

using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace CSharp.AsposePDFFacades.Attachments
{
    public class ExtractAllAttachments
    {
        public static void Run()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_AsposePdfFacades_Attachments();
            //open document
            PdfExtractor pdfExtractor = new PdfExtractor();
            pdfExtractor.BindPdf(dataDir+ "ExtractAllAttachments.pdf");

            //extract attachments
            pdfExtractor.ExtractAttachment();

            //get extracted attachments
            pdfExtractor.GetAttachment(dataDir+ ".\\output");
            
            
        }
    }
}

File: Examples/VisualBasic/AsposePdfFacades/Attachments/ExtractAllAttachments.vb
Imports System
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Pdf
Imports Aspose.Pdf.Facades

Namespace AsposePDFFacades.Attachments
    Public Class ExtractAllAttachments
        Public Shared Sub Run()
            ' ExStart:ExtractAllAttachments
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_AsposePdfFacades_Attachments()
            ' Open document
            Dim pdfExtractor As New PdfExtractor()
            pdfExtractor.BindPdf(dataDir & "ExtractAllAttachments.pdf")

            ' Extract attachments
            pdfExtractor.ExtractAttachment()

            ' Get extracted attachments
            pdfExtractor.GetAttachment(dataDir)
            ' ExEnd:ExtractAllAttachments
        End Sub
    End Class
End Namespace